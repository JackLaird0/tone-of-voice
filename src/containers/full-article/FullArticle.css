.container {
  height: 760px;
  width: 900px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: rgba(255, 255, 255, 0.356);
  margin: 30px auto;
  border-radius: 5px;
}

.article-title {
  font-size: 20px;
  padding: 0 40px;
  color: rgb(24, 24, 24);
}

.article-author {
  font-size: 16px;
  padding: 0 40px;
}

.text-container {
  /* box-shadow: 0px 4px 4px rgb(92, 92, 92); */
  height: 500px;
  width: 860px;
  overflow: scroll;
  /* padding: 20px; */
  margin: 20px 10px;
}

.loading {
  margin: 100px auto;
  display: block;
}

.article-text {
  padding: 0 40px;
  line-height: 26px;
  color: rgb(24, 24, 24);
}

.image {
  height: 270px;
  width: 370px;
  margin-top: 20px;
  border-radius: 5px;
}

.article-link {
  margin: 16px 0 30px 0;
  height: 40px;
  width: 120px;
  background-color: #5d748c;
  color: white;
  text-decoration: none;
  border-bottom: solid 2px #485d74;
  box-shadow: 0px 3px 3px grey;
  text-align: center;
  padding-top: 10px;
  border-radius: 5px;
  transition: 0.5s;
}

.article-link:hover {
  background-color: #485d74;
  box-shadow: 0px 1px 3px grey;
  border-bottom: solid 2px #485d7400;
}