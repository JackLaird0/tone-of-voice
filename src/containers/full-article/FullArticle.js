import React, { Component } from 'react';
import { connect } from 'react-redux'

export class FullArticle extends Component {

  componentDidMount() {
    if(!this.props.article.text) {
      this.props.history.push('/')
    }
  }

  render () {
    const {title, author, text, pageUrl} = this.props.article;
    return (
      <div>
        <h2>{title}</h2>
        <h5>{author}</h5>
        <p>{text}</p>
        <a href={pageUrl} target='_blank'> View Article</a>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  article: state.article
})

export default connect(mapStateToProps)(FullArticle)