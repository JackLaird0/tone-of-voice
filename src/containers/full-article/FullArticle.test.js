import { FullArticle, mapStateToProps } from './FullArticle';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import React from 'react';

describe('Header', () => {

  let mockProps;
  let fullArticle;

  beforeEach(() => {

    mockProps = {
      article: {
        title: 'trump',
        author: 'me',
        text: 'interesting text',
        pageUrl: 'coolarticle.io',
        images: 'coolarticle-image.io'
      }
    };

    fullArticle = shallow(<FullArticle {...mockProps} />, { disableLifecycleMethods: true });
  });

  describe('component', () => {

    it('matches snapshot', () => {
      fullArticle = shallow(<FullArticle {...mockProps} />, { disableLifecycleMethods: true });
      expect(fullArticle).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {

    it('returns an object with news', () => {
      let article = mockProps.article;

      const mockState = {
        article
      };
  
      const mappedProps = mapStateToProps(mockState);
  
      expect(mappedProps).toEqual(mockState);
    });
  });
});