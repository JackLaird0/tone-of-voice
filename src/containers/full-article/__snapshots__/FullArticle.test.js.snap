// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FullArticle
    article={
        Object {
            "author": "me",
            "pageUrl": "coolarticle.io",
            "text": "interesting text",
            "title": "trump",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          trump
</h2>,
        <h5>
          me
</h5>,
        <p>
          interesting text
</p>,
        <a
          href="coolarticle.io"
          target="_blank"
>
           View Article
</a>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "trump",
        },
        "ref": null,
        "rendered": "trump",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "me",
        },
        "ref": null,
        "rendered": "me",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "interesting text",
        },
        "ref": null,
        "rendered": "interesting text",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " View Article",
          "href": "coolarticle.io",
          "target": "_blank",
        },
        "ref": null,
        "rendered": " View Article",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            trump
</h2>,
          <h5>
            me
</h5>,
          <p>
            interesting text
</p>,
          <a
            href="coolarticle.io"
            target="_blank"
>
             View Article
</a>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "trump",
          },
          "ref": null,
          "rendered": "trump",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "me",
          },
          "ref": null,
          "rendered": "me",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "interesting text",
          },
          "ref": null,
          "rendered": "interesting text",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " View Article",
            "href": "coolarticle.io",
            "target": "_blank",
          },
          "ref": null,
          "rendered": " View Article",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
